src/main/java/io/nats/streaming/StreamingConnectionImpl.java

```diff
@@ -552,14 +552,25 @@ protected void processAck(Message msg) {
     TimerTask createAckTimerTask(String guid) {
         TimerTask task = new java.util.TimerTask() {
             public void run() {
-                processAckTimeout(guid);
+                try {
+                    processAckTimeout(guid);
+                } catch (Exception e) {
+                    // catch exception to prevent the timer to be closed
+                    logger.error("stan: error encountered during processAckTimeout, will cancel this timer task", e);
+                    // cancel this task
+                    cancel();
+                }
             }
         };
         return task;
     }
 
     protected void processAckTimeout(String guid) {
         AckClosure ackClosure = removeAck(guid);
+        if (ackClosure == null) {
+            logger.warn("stan: no ack available, could be due to the race condition");
```

Comment:
Not sure we need to log anything here. The "race" is not the result of a bug but a fact of life: the timer may expire and fire the callback at the same time that the ack is received and processed on another thread. We need to not process it twice for sure by returning here, but no need to log.